TreeDB/cmd/treedb_rag_interleave/main.go

```
@@ -0,0 +1,459 @@
+		err := worker.cmd.Wait()
+		worker.evidence.FinishedAt = time.Now().UTC()
+		if err != nil && firstErr == nil {
+			firstErr = fmt.Errorf("wait for %s: %w: %s", worker.cfg.Name, err, strings.TrimSpace(worker.stderr.String()))
+		}
+	}
+	return firstErr
+}
+
+func writeArtifact(path string, value artifact) (err error) {
+	dir := filepath.Dir(path)
+	if err := os.MkdirAll(dir, 0o755); err != nil {
```

golangci-lint (2.12.2)

[medium] G301: Expect directory permissions to be 0750 or less

(gosec)